server/routes/testutils/appSetup.ts

```diff
@@ -31,30 +32,50 @@ class MockUserService extends UserService {
   }
 }
 
-function appSetup(route: Router, production: boolean): Express {
+function appSetup({
+  router,
+  userRoles = [AuthRole.OMU],
+  production = false,
+}: {
+  router: Router
+  userRoles?: AuthRole[]
+  production?: boolean
+}): Express {
   const app = express()
 
   app.set('view engine', 'njk')
 
   nunjucksSetup(app, path)
 
   app.use((req, res, next) => {
-    res.locals = {}
-    res.locals.user = req.user
+    req.user = { ...user, token: 'token1', authSource: 'nomis', username: 'user1' }
+    res.locals = { user: { ...req.user, userRoles } }
     next()
   })
 
   app.use(cookieSession({ keys: [''] }))
   app.use(express.json())
   app.use(express.urlencoded({ extended: true }))
-  app.use('/', route)
+  app.use('/', router)
   app.use((req, res, next) => next(createError(404, 'Not found')))
   app.use(errorHandler(production))
 
   return app
 }
 
-export default function appWithAllRoutes({ production = false }: { production?: boolean }): Express {
+// eslint-disable-next-line import/prefer-default-export
+export const appWithAllRoutes = (
+  overrides: Partial<Services> = {},
+  userRoles = [AuthRole.OMU],
+  production?: boolean
+): Express => {
+  const router = allRoutes({
+    userService: new MockUserService(),
+    prisonerService: {} as PrisonerService,
+    licenceService: {} as LicenceService,
+    communityService: {} as CommunityService,
+    ...overrides,
+  })
   auth.default.authenticationMiddleware = () => (req, res, next) => next()
-  return appSetup(allRoutes(standardRouter(new MockUserService())), production)
+  return appSetup({ router, userRoles, production })
 }
```

Comment:
I'm not entirely happy with this `appSetup()` for unit testing yet. But it works for now, and I'll continually improve as and when I see issues.